src/devices/bus/pce_ctrl/joypad2.cpp

```diff
@@ -0,0 +1,136 @@
+// license:BSD-3-Clause
+// copyright-holders:cam900
+/**********************************************************************
+
+    NEC PC Engine/TurboGrafx-16 2 button joypad emulation
+
+    Based on SMS controller port emulation (devices\bus\sms_ctrl\*.*)
+	by Fabio Priuli,
+	PC engine emulation (mame\*\pce.*)
+	by Charles MacDonald, Wilbert Pol, Angelo Salese
+
+    First party models:
+
+    NEC PC Engine Pad (PI-PD001)
+    No autofire, Bundled in PC Engine
+
+    NEC Turbo Pad (PI-PD002 and Later models)
+    Add autofire support
+
+    NEC Turbo Stick (PI-PD4)
+    Arcade joystick variation of Turbo Pad, with Seperated autofire
+	buttons.
+    Turbo Stick for TurboGrafx-16 has slow motion.
+
+    Everything else PI-PD001 supports Autofire.
+
+    TODO:
+	- Needs verifications for Part numbers
+
+**********************************************************************/
+
+#include "emu.h"
+#include "joypad2.h"
+
+
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(PCE_JOYPAD2, pce_joypad2_device, "pce_joypad2", "NEC PC Engine/TurboGrafx-16 2 Button Joypad")
+
+
+static INPUT_PORTS_START( pce_joypad2 )
+	// II is left of I on the original pad so we map them in reverse order
+	PORT_START("BUTTONS")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_NAME("Button I")
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_NAME("Button II")
```

Comment:
Why reverse the order?  It would make more sense to make I button 1 and II button 2.

Reply:
Same reason why NES buttons on MAME (or any emulator) are ordered B,A.

Reply:
It makes no sense with Avenue Pad 3 and Avenue Pad 6 when you get additional buttons if it goes II, I, III or whatever.

Reply:
Historically IPT_BUTTON1 is leftmost in a row, and a 2b PCE pad sports a II/I layout. i.e. Most games maps jump/bomb to I while fire is II.
3b/6b maps the row as III/II/I -> b1/b2/b3 supposedly.

Reply:
It’s still better not to have the buttons jump around when you go to a pad with more buttons.